Src/LogData.c

```diff
@@ -230,29 +234,55 @@ void logDataTask(void const* arg)
 
     for (;;)
     {
-        switch (getCurrentFlightPhase())
-        {
-            case PRELAUNCH:
-                lowFrequencyLogToSdRoutine(data, buffer, PRELAUNCH);
-                break;
-
-            case BURN:
-            case COAST:
-            case DROGUE_DESCENT:
-                highFrequencyLogToSdRoutine(data, buffer);
-                break;
-
-            case MAIN_DESCENT:
-                lowFrequencyLogToSdRoutine(data, buffer, MAIN_DESCENT);
-                break;
-
-            case ABORT:
-                lowFrequencyLogToSdRoutine(data, buffer, ABORT);
-                break;
-
-            default:
-                lowFrequencyLogToSdRoutine(data, buffer, MAIN_DESCENT); // won't work
-                break;
+        checkForHeader();
+        if(containsHeader == 1){
+            switch (getCurrentFlightPhase())
+            {
+                case PRELAUNCH:
+                    lowFrequencyLogToSdRoutine(data, buffer, PRELAUNCH);
+                    break;
+
+                case BURN:
+                case COAST:
+                case DROGUE_DESCENT:
+                    highFrequencyLogToSdRoutine(data, buffer);
+                    break;
+
+                case MAIN_DESCENT:
+                    lowFrequencyLogToSdRoutine(data, buffer, MAIN_DESCENT);
+                    break;
+
+                case ABORT:
+                    lowFrequencyLogToSdRoutine(data, buffer, ABORT);
+                    break;
+
+                default:
+                    lowFrequencyLogToSdRoutine(data, buffer, MAIN_DESCENT); // won't work
+                    break;
+            }    
+        }else if(containsHeader ==  0){
+            //Open file
+            f_open(&file, fileName, FA_CREATE_NEW | FA_READ | FA_WRITE);
+            //Write header
+            f_puts(buffer, &file);
+            //Make sure header is written to file
+            checkForHeader();
         }
     }
 }
+
+//This function checks if there's a header
+void checkForHeader(){
+    //open file again then read if theres a header
+    f_open(&file, fileName, FA_CREATE_NEW | FA_READ | FA_WRITE);
+    //if the file is empty, keep trying to write header.
+    if(f_size(&file) == 0){
```

Comment:
Since containsHeader starts at 0, we can replace this ```if else``` statement with a single if statement.
```
if(f_size(&file))
{
    containsHeader = 1;
}
```
This will only change containsHeader from 0 to 1 if the header is written.